File: module_other/src/main/java/com/ashlikun/baseproject/module/other/mode/ApiOther.java
package com.ashlikun.baseproject.module.other.mode;

import com.ashlikun.baseproject.libcore.utils.http.BaseApiService;

/**
 * 作者　　: 李坤
 * 创建时间: 2017/8/10　9:31
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：请求mode类
 */

public class ApiOther extends BaseApiService {
    private static ApiOther api;

    public static ApiOther getApi() {
        if (api == null) {
            synchronized (ApiOther.class) {
                if (api == null) {
                    api = new ApiOther();
                }
            }
        }
        return api;
    }
}

File: module_other/src/main/java/com/ashlikun/baseproject/module/other/mode/ApiOther.kt
package com.ashlikun.baseproject.module.other.mode

import com.ashlikun.baseproject.libcore.mode.ApiBase
import com.ashlikun.baseproject.libcore.utils.http.HttpUiHandle
import com.ashlikun.baseproject.libcore.utils.http.requestGet
import com.ashlikun.baseproject.libcore.utils.http.syncExecute
import com.ashlikun.okhttputils.http.response.HttpResponse
import com.ashlikun.okhttputils.retrofit.Retrofit


/**
 * 作者　　: 李坤
 * 创建时间: 2017/8/10　9:31
 * 邮箱　　：496546144@qq.com
 *
 *
 * 功能介绍：请求mode类
 */

interface ApiOther : ApiBase {
    companion object {
        val api: ApiOther by lazy { Retrofit.get().create(ApiOther::class.java) }
    }

    suspend fun testx(handle: HttpUiHandle
    ): HttpResponse? {
        return "index".requestGet()
                .syncExecute(handle)
    }


}
